File: server/models/Doctor.js
const mongoose = require('mongoose');

const doctorSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  specialtyId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Specialty',
    required: true
  },
  hospitalId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Hospital',
    required: true
  },
  title: {
    type: String,
    required: [true, 'Chức danh là bắt buộc'],
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  education: {
    type: String,
    trim: true
  },
  experience: {
    type: Number, // Số năm kinh nghiệm
    default: 0
  },
  certifications: [{
    type: String,
    trim: true
  }],
  languages: [{
    type: String,
    trim: true
  }],
  consultationFee: {
    type: Number,
    required: true,
    default: 0
  },
  isAvailable: {
    type: Boolean,
    default: true
  },
  ratings: {
    average: {
      type: Number,
      default: 0
    },
    count: {
      type: Number,
      default: 0
    }
  },
  averageRating: {
    type: Number,
    default: 0
  },
  reviews: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Review'
  }]
}, {
  timestamps: true
});

// Virtual field for full name
doctorSchema.virtual('fullInfo').get(function() {
  return `${this.title} ${this.user.fullName}`;
});

// Ensure virtuals are included in JSON
doctorSchema.set('toJSON', { virtuals: true });
doctorSchema.set('toObject', { virtuals: true });

const Doctor = mongoose.model('Doctor', doctorSchema);

module.exports = Doctor; 

File: hospital_mobile_app/lib/domain/entities/doctor.dart
import 'package:equatable/equatable.dart';

class Doctor extends Equatable {
  final String id;
  final String fullName;
  final String email;
  final String? avatar;
  final String specialtyId;
  final String specialtyName;
  final String? bio;
  final int experience;
  final String? education;
  final List<String> languages;
  final double rating;
  final int reviewCount;
  final double consultationFee;
  final bool isAvailable;
  final DateTime createdAt;

  const Doctor({
    required this.id,
    required this.fullName,
    required this.email,
    this.avatar,
    required this.specialtyId,
    required this.specialtyName,
    this.bio,
    required this.experience,
    this.education,
    required this.languages,
    required this.rating,
    required this.reviewCount,
    required this.consultationFee,
    required this.isAvailable,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        fullName,
        email,
        avatar,
        specialtyId,
        specialtyName,
        bio,
        experience,
        education,
        languages,
        rating,
        reviewCount,
        consultationFee,
        isAvailable,
        createdAt,
      ];
}
